### PR TITLE
feat(settings): Verify password

### LINKED ISSUE #4527: Double verify new password when changing password
Right now, there's only one field for "New Password". If you quickly typed a password wrong, there's no way to really know since we don't require a double verification.

Maybe we should tuck this away under a "Change Password" CTA to avoid the clutter?

/cc @kjlundsgaard @ckj

### PR DESCRIPTION
**Pull request by lauryndbrown**
When the user changes their password in the Sentry app require that they type it twice to verify that their new password is the password they wanted.

Fixes  #4527

### CODE REVIEW

**Review by benvinegar — CHANGES_REQUESTED** [review]
Please add a test that demonstrates behavior when `verify_new_password` and `new_password` differ (e.g. returns a page with an error).

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: tests/sentry/web/frontend/accounts/tests.py [hunk 1]
```diff
@@ -212,6 +215,62 @@ def test_cannot_change_email_with_invalid_password(self):
...
+        params['new_password'] = 'foobar'
+
+        resp = self.client.post(self.path, params)
+        assert resp.status_code == 200
+        self.assertTemplateUsed('sentry/account/settings.html')
+        assert resp.context['form'].errors
+        user = User.objects.get(id=self.user.id)
+        assert not user.check_password('foobar')
+
+    def test_cannot_change_password_with_unequal_verify_password(self):
```
Comment (lauryndbrown) [inline tests/sentry/web/frontend/accounts/tests.py]: @benvinegar I added tests for checking if there is no verify password and if new_password and verify_password are unequal. Let me know if these look ok.

**Comment by benvinegar** [comment]
Last thing. I'm a little concerned that this is confusing:



Can we re-order the fields and heading to match what I mocked in #4527 (break out "change password" into its own section)? Should only take a few minutes, I can help.

**Comment by lauryndbrown** [comment]
@benvinegar Of course! That was an oversight on my end. Let me work on that

**Comment by lauryndbrown** [comment]
screen shot 2017-12-12 at 4 30 45 pm

**Comment by lauryndbrown** [comment]
@benvinegar were you wanting it like this or full width?

**Comment by lauryndbrown** [comment]
The above is the effect of putting it inside the div with the account-settings-body class. I can create another div and create a full width one instead. Just let me know what you might like. I wasn't sure looking at the mockup

**Comment by benvinegar** [comment]
Is password required for changing email? (Should be if it isn't.)

**Comment by lauryndbrown** [comment]
@benvinegar So, I'm confused as to what is the desired behavior here. The UI states that "You will need to enter your current account password to make changes." However, when I change a name or an email, I can without a password. Would you like me to remove that in the UI? I think that would be confusing. Then I can just change the UI to exactly your mockup.

File: tests/sentry/web/frontend/accounts/tests.py (hunk 1 above)
Comment (benvinegar) [reply tests/sentry/web/frontend/accounts/tests.py]: They look good!

**Review by benvinegar — APPROVED** [review]
Okay, this looks good. There are some usability issues with these pages, but I think those are outside the scope of this PR, which adds an objective security improvement.

**Comment by benvinegar** [comment]
💯

**Pull request merged**